hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -207,11 +195,15 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         updateHiveTable = true;
         LOG.debug("Committing existing table: {}", fullName);
       } else {
-        tbl = newHmsTable(metadata);
+        tbl =
+            newHmsTable(
+                metadata.property(HiveCatalog.HMS_TABLE_OWNER, HiveHadoopUtil.currentUser()));
```

Comment:
I am really sorry for missing this before.

Why do we set this generally in the `newHmsTable`?

Reply:
Yes, it was fetching [tableOwner](https://github.com/apache/iceberg/blob/229a243ddad20a725d9f7e4446ce8bb3fc2c9a53/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L369) from TableMetadata.

Reply:
Don't we have the same `tableOwner` for views as well?

Reply:
For views, I thought it's always going to be [CurrentUser](https://github.com/apache/iceberg/pull/8907/files#diff-db46657b84d66e084e15f31b8dab21577efb2ae7102863f94c6c9477782de676R320). But I think it makes sense to  get the tableOwner from ViewMetadata as well if Present.

Reply:
You might be better positioned to suggest things from the `View` point of view 😄, but if it were only me, then I would have created the  following method in `HiveOperationsBase`:
```
default Table newHmsTable(TableMetadata metadata) {
[..]
}
```
and kept the implementation the same (using the same metadata key to get the owner), like before:
```
    Preconditions.checkNotNull(metadata, "'metadata' parameter can't be null");
    final long currentTimeMillis = System.currentTimeMillis();

    Table newTable =
        new Table(
            tableName,
            database,
            metadata.property(HiveCatalog.HMS_TABLE_OWNER, HiveHadoopUtil.currentUser()),
            (int) currentTimeMillis / 1000,
            (int) currentTimeMillis / 1000,
            Integer.MAX_VALUE,
            null,
            Collections.emptyList(),
            Maps.newHashMap(),
            null,
            null,
            TableType.EXTERNAL_TABLE.toString());

    newTable
        .getParameters()
        .put("EXTERNAL", "TRUE"); // using the external table type also requires this
    return newTable;
```

Reply:
[TableMetadata](https://github.com/apache/iceberg/blob/c1190990760509e40dd8e10092196a569d684f1b/core/src/main/java/org/apache/iceberg/TableMetadata.java)  and [ViewMetadata](https://github.com/apache/iceberg/blob/c1190990760509e40dd8e10092196a569d684f1b/core/src/main/java/org/apache/iceberg/view/ViewMetadata.java) both are unrelated. I am not able to find a better approach with `newHmsTable(TableMetadata metadata) `.  
Please suggest.

Reply:
Probably it would be the best to have a common interface for the 2, but we should not hold back this part for it.